packages/backend/src/modules/tvl/ChainTvlSubmodule.ts

```diff
@@ -0,0 +1,117 @@
+import { HttpClient, Logger, UniversalEtherscanClient } from '@l2beat/shared'
+import { ChainId, ProjectId, Token } from '@l2beat/shared-pure'
+import { providers } from 'ethers'
+
+import { ChainTvlConfig } from '../../config/Config'
+import { TotalSupplyFormulaUpdater } from '../../core/assets/TotalSupplyFormulaUpdater'
+import { BlockNumberUpdater } from '../../core/BlockNumberUpdater'
+import { Clock } from '../../core/Clock'
+import { PriceUpdater } from '../../core/PriceUpdater'
+import { TotalSupplyProvider } from '../../core/totalSupply/TotalSupplyProvider'
+import { TotalSupplyUpdater } from '../../core/totalSupply/TotalSupplyUpdater'
+import { EthereumClient } from '../../peripherals/ethereum/EthereumClient'
+import { MulticallClient } from '../../peripherals/ethereum/multicall/MulticallClient'
+import { MulticallConfigEntry } from '../../peripherals/ethereum/multicall/types'
+import { TvlSubmodule } from '../ApplicationModule'
+import { TvlDatabase } from './types'
+
+export function chainTvlSubmodule(
+  name: string,
```

Comment:
```suggestion
  submoduleName: string,
```

Reply:
or we can skip it alltogether and just use ChainId.getName() to create this string